pkg/deployments/tasks/20230217-gnosis-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,326 @@
+    await task.run({ force: true });
+    factory = await task.deployedInstance('GnosisRootGaugeFactory');
+  });
+
+  before('advance time', async () => {
+    // This causes all voting cooldowns to expire, letting the veBAL holder vote again
+    await advanceTime(DAY * 12);
+  });
+
+  before('setup accounts', async () => {
+    admin = await getSigner(0);
+    recipient = await getSigner(1);
+
+    veBALHolder = await impersonate((await getSigner(2)).address, VAULT_BOUNTY.add(fp(5))); // plus gas
+  });
+
+  before('setup contracts', async () => {
+    const vaultTask = new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    vault = await vaultTask.deployedInstance('Vault');
+    authorizer = await vaultTask.instanceAt('Authorizer', await vault.getAuthorizer());
+
+    const authorizerAdaptorTask = new Task('20220325-authorizer-adaptor', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    authorizerAdaptor = await authorizerAdaptorTask.deployedInstance('AuthorizerAdaptor');
+
+    const gaugeAdderTask = new Task('20230109-gauge-adder-v3', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    gaugeAdder = await gaugeAdderTask.deployedInstance('GaugeAdder');
+
+    const balancerTokenAdminTask = new Task('20220325-balancer-token-admin', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    BALTokenAdmin = await balancerTokenAdminTask.deployedInstance('BalancerTokenAdmin');
+
+    BAL = await BALTokenAdmin.getBalancerToken();
+
+    const gaugeControllerTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    gaugeController = await gaugeControllerTask.deployedInstance('GaugeController');
+  });
+
+  before('update balancer token admin rate', async () => {
+    await advanceTime(WEEK * 5);
+    await BALTokenAdmin.update_mining_parameters();
+  });
+
+  before('create veBAL whale', async () => {
+    const veBALTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    veBAL = await veBALTask.instanceAt('VotingEscrow', veBALTask.output({ network: 'mainnet' }).VotingEscrow);
+
+    bal80weth20Pool = await deployedAt('v2-pool-weighted/WeightedPool', VEBAL_POOL);
+
+    const poolId = await bal80weth20Pool.getPoolId();
+
+    await vault.connect(veBALHolder).joinPool(
+      poolId,
+      veBALHolder.address,
+      veBALHolder.address,
+      {
+        assets: [BAL, ZERO_ADDRESS],
+        maxAmountsIn: [0, VAULT_BOUNTY],
+        fromInternalBalance: false,
+        userData: WeightedPoolEncoder.joinExactTokensInForBPTOut([0, VAULT_BOUNTY], 0),
+      },
+      { value: VAULT_BOUNTY }
+    );
+
+    await bal80weth20Pool.connect(veBALHolder).approve(veBAL.address, MAX_UINT256);
+    const currentTime = await currentTimestamp();
+    await veBAL
+      .connect(veBALHolder)
+      .create_lock(await bal80weth20Pool.balanceOf(veBALHolder.address), currentTime.add(MONTH * 12));
+  });
```

Comment:
This is all a bit convoluted - you can just impersonate any veBAL holder and use their address, and you'll skip having to join, lock, etc.

Reply:
It was not trivial to find one with enough veBAL to make the voting work, and it also constrains it to be at a certain block. This way we don't have to worry about that, and can do it any time.